Add support for offline subtitles checking for custom language title

burekas7 opened the pull request:
**Add support for offline subtitles checking for episodes for custom language TV Show title**

When the Tv Show title is in hebrew (By TVDB scraper for example)
The query is like this
ווסטוורלד S02E03
And it can't get results because of this.
So I add a function which translate the tv show title to English using TMDB api.
Tested for Hebrew.

burekas7 committed:
Add support for tv shows episodes offline subtitles checking for custom language title

Add support for offline subtitles checking for custom language title

When the Tv Show title is in hebrew (By TVDB scraper for example)
The query is like this
ווסטוורלד S02E03
And it can't get results because of this.
So I add a function which translate the tv show title to English using TMDB api.
Tested for Hebrew.

burekas7 committed:
Add support for offline TV Show episodes subtitles checking for custom language title

Add support for offline TV Show episodes subtitles checking for custom language title

When the Tv Show title is in hebrew (By TVDB scraper for example)
The query is like this
ווסטוורלד S02E03
And it can't get results because of this.
So I add a function which translate the tv show title to English using TMDB api.
Tested for Hebrew.

burekas7 committed:
Merge pull request #2 from burekas7/burekas7-OfflineTvShowCustomTitle-1

Add support for offline TV Show episodes subtitles checking for custo…

burekas7 committed:
Merge pull request #1 from burekas7/burekas7-OfflineTvShowCustomTitle

Add support for tv shows episodes offline subtitles checking for cust…

burekas7 committed:
Update service.py

burekas7 committed:
Merge pull request #3 from burekas7/burekas7-OfflineTvShowCustomTitle

Update service.py

burekas7 committed:
Update SUBUtilities.py

CaTzil commented on resources/lib/SUBUtilities.py:
```diff
@@ -260,3 +266,58 @@ def request(self, url, data=None, query_string=None, referrer=None, cookie=None)
+def lowercase_with_underscores(str):
+	return normalize('NFKD', unicode(unicode(str, 'utf-8'))).encode('utf-8', 'ignore')
+
+def cachingJSON(filename, url):
```
methods in python should be snake case.

burekas7 replied:
Fixed.

CaTzil commented on resources/lib/SUBUtilities.py:
```diff
@@ -260,3 +266,58 @@ def request(self, url, data=None, query_string=None, referrer=None, cookie=None)
+			"tv",tmdbKey, labelTVShowTitle)
+    
+    log("searchTMDB: %s" % url)
+    json = cachingJSON(filename,url)
```
What is the purpose of this cache?

burekas7 replied:
Basically this code was taken from another place.
But as I understand it saves communication with the tmdb api if you've already converted to the same series name before, because the json data file already exist in the addon data folder.

CaTzil commented on service.py:
```diff
@@ -144,7 +129,7 @@ def collect_initial_data():
         item_data['season'] = xbmc.getInfoLabel("ListItem.Season")
         item_data['episode'] = xbmc.getInfoLabel("ListItem.Episode")
-        item_data['tvshow'] = xbmc.getInfoLabel("ListItem.TVShowTitle")
+        item_data['tvshow'] = getTVshowOriginalTitleByTMDBapi()
```
It is better to pass params to this method instead of accessing global `xbmc` inside it?

CaTzil replied:
It this method works only for TVShows? what about movies?

burekas7 replied:
1. I don't think it makes any different. Keep it as it is.
If you think it's better calling them outside I will change that. This is your decision.

2. Regarding movies, the ListItem.TVShowTitle"" always returns the title in english, so there is no problem here as tv shows when "ListItem.TVShowTitle" return the title in Hebrew (for example) and "ListItem.TVShowTitle" returns empty string, so in order to make a string query the only option is to convert the Hebrew title to English (As implemented in this PR)
* Long time ago I could get the original title in English from kodi json api, but now it's not working anymore (probably changes in TVDB scraper or something), I reported on that.
* But I think it depends on the Movies scraper, mine is "Universal Movie Scraper".

CaTzil commented on resources/lib/SUBUtilities.py:
```diff
@@ -260,3 +266,58 @@ def request(self, url, data=None, query_string=None, referrer=None, cookie=None)
+
+    return title
+
+def getTVshowOriginalTitleByTMDBapi():
```
Where is the check if it is not english?
Why add another request & all of this logic for users that have english titles?
What about "regular" users that has titles in Hebrew?

burekas7 replied:
How can I check if title is in English?
If it's possible I will add a condition here, but I don't know how.

It was tested with Hebrew and English titles.

CaTzil commented on resources/lib/SUBUtilities.py:
```diff
@@ -10,6 +10,12 @@
 import xbmcvfs
 import xbmcaddon
 
+from unicodedata import normalize
```
did you checked this plugin in Kodi 19? Cause sagi worked hard to make it work on it.

burekas7 replied:
Basically I'm still working only with kodi 17.6
But I can ask other users to test it if needed.
I guess it will work with kodi 19.

And as I said, this PR only relates to the Offline Subtitles Checking Support.

CaTzil replied:
Pls check it on Kodi 19, cause even wrong import will throw exception (check what Sagi have changed in his PR).

burekas7 replied:
@sagiben 
Sagi, can you test this PR for me with Kodi 19 (python 3)?

@CaTzil 
I have understood that you have Kodi 18 (python 2.7 & 3), so tell me if you can test it with your version too.

Thanks.

Those was added for this PR:
```
from unicodedata import normalize
from os import path
from requests import get
from json import loads, load
from time import time
```

sagiben replied:
Hi, What exactly is the test case ?
I used plugin version 1.2.0 w/o the PR a TVShow with an Hebrew filename and was able to find subtitles for that tvshows automatically

burekas7 replied:
@sagiben 
Basically at first to check if using the addon throwing an exception due to wrong or unsupported imports, and second to test what you have already checked but in Kodi Library that scrap tv shows with Hebrew titltes (It's strange a little that it works for you even without the PR, maybe in Kodi 19 it's different)

But it seems that it's ok according to your comment. Thanks.

CaTzil commented:
I've sent a feature request to ScrewZira to add support for Hebrew search.
I think that this is not plugin's job to add this kind of support.

burekas7 committed:
Update SUBUtilities.py

Update SUBUtilities.py according to the comment in PR 4,
Change function name to snake case pattern.

burekas7 commented:
> I've sent a feature request to ScrewZira to add support for Hebrew search.
> I think that this is not plugin's job to add this kind of support.

Makes sense, if they will accept to do this.
Just pay attention that it should support both Hebrew and English.

But if they won't, this method works perfect for me.
